File: Algorithm_83_Remove Duplicates from Sorted List.cpp
#include <iostream>
using namespace std;
struct ListNode {
	int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
}; 
ListNode* deleteDuplicates(ListNode* head)
{
	ListNode *tmp=head;
	//ListNode *deleted;
	while (tmp!=NULL)
	{
		if (tmp->next!=NULL && tmp->next->val==tmp->val)
		{
			//deleted=tmp->next;
			tmp->next=tmp->next->next;
			//delete deleted;
		}
		else tmp=tmp->next;
	}
	return head;
}

File: Algorithm_83_Remove Duplicates from Sorted List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */

// 思路：Straight-forward

class Solution {
    public ListNode deleteDuplicates(ListNode head) {
        ListNode dummy = new ListNode(0);
        ListNode newCur = dummy, cur = head;
        while (cur != null) {
            while (cur.next != null && cur.next.val == cur.val) {
                cur = cur.next;
            }
            newCur.next = cur;
            cur = cur.next;
            newCur = newCur.next;
        }
        newCur.next = null;
        return dummy.next;
    }
}